test/framework/toy_build.py

```diff
@@ -1194,6 +1194,40 @@ def test_toy_extension_patches(self):
 
         self.test_toy_build(ec_file=test_ec)
 
+    def test_toy_extension_git(self):
+        """Test install toy that includes extensions with git_config."""
+        test_ecs = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'easyconfigs', 'test_ecs')
+        toy_ec = os.path.join(test_ecs, 't', 'toy', 'toy-0.0.eb')
+        toy_ec_txt = read_file(toy_ec)
+
+        # Tar-ball which should be created via 'git_config'
+        ext_tgz = 'exts-git.tar.gz'
+        ext_tarball = os.path.join(self.test_sourcepath, 't', 'toy', ext_tgz)
+
+        # Dummy source code required for extensions build_step to pass
+        ext_src = 'int main() { return 0; }'
+        ext_cfile = 'exts-git.c'
+
+        test_ec = os.path.join(self.test_prefix, 'test.eb')
+        test_ec_txt = '\n'.join([
+            toy_ec_txt,
+            'prebuildopts = "echo \\\"%s\\\" > %s && ",' % (ext_src, ext_cfile),
+            'exts_list = [',
+            '   ("exts-git", "0.0", {',
+            '       "buildopts": "&& ls -l %s",' % ext_tarball,
+            '       "source_tmpl": "%s",' % ext_tgz,
+            '       "git_config": {',
```

Comment:
For non-extensions, `git_config` is a way of specifying an entry in `sources`.
Should we do that here too?

I think long-term we want to move away from the `source_tmpl` (which is custom to extensions), and move to using `sources` for extensions as well. For now, we can restrict it to a single-element list for an extension, we've never run into a case where an extension requires multiple sources afaik.

So, we should:

* introduce support for `sources` in an extension (which should have preference over `source_tmpl` in case both are specified);
* support `git_config` as a part of `sources`, like:
  ```
  exts_list = [
    ('exts-git', '0.0', {
        'sources': [{
            'filename': 'exts-git-0.0.tar.gz',
            'git_config': {
                ...
            }],
    }),
  ]
  ```

Let me know if you're up for this...

Reply:
This needs some thinking.  I imagine it would require continuing to support the existing interpretation of `extra_options`, and having an alternative if-block where `extra_options` contains only `sources`.

Perhaps the expansion of `sources` could be encapsulated and just called from here?

Reply:
Okay, breakfast and lunch gave me time to digest this more.  I'm thinking that since there's at least one other user out there (#3251) who would like this functionality, it might serve the community better to roll out the functionality here, and open a separate Issue or PR to support the nested `extra_options = { 'sources': [{...}] }` functionality.  

I realize that means `git_config` is apparently being handled in two different ways, but that's not true behind the scenes.  In both cases, the `git_config` dictionary value is captured and passed through to `obtain_file`, without being separately parsed, so the behaviour would be consistent.

Reply:
Following up, since it's showing that there are still "unresolved changes" from the review, but I can't find them.

I have successfully (I think!) implemented putting a full 'sources' block inside an 'exts_list' entry.  I did not remove the bare 'git_config' entry in 'exts_list', because it already worked, and because (I think) each 'exts_list' is supposed to represent a single source, not a complex multisource compilation.

This does mean, as I noted above, that this gives two syntactic ways of achieving the same end result.  If you would like the bare 'git_config' support removed, please let me know.